src/zeta/aca_zeta_programming.cpp

```diff
@@ -181,34 +190,62 @@ int ACA_Zeta_Programming::create_zeta_config(const alcor::schema::AuxGateway cur
 {
   ACA_LOG_DEBUG("%s", "ACA_Zeta_Programming::create_zeta_config ---> Entering\n");
   int overall_rc = EXIT_SUCCESS;
-  zeta_config *new_zeta_cfg;
+  zeta_config *current_zeta_cfg;
   bool bucket_not_found = false;
-  unordered_set<string> new_zeta_buckets;
+  CTSL::HashMap<fwd_info *, int *> new_zeta_buckets;
 
   uint oam_port = current_AuxGateway.zeta_info().port_inband_operation();
 
-  if (!_zeta_config_table.find(current_AuxGateway.id(), new_zeta_cfg)) {
-    create_entry(current_AuxGateway.id(), oam_port);
+  if (!_zeta_config_table.find(current_AuxGateway.id(), current_zeta_cfg)) {
+    create_entry(current_AuxGateway.id(), oam_port, current_AuxGateway);
+    _zeta_config_table.find(current_AuxGateway.id(), current_zeta_cfg);
+    
+    //-----Start unique lock-----
+    std::unique_lock<std::timed_mutex> group_entry_lock(_group_operation_mutex);
+    overall_rc = _create_zeta_group_entry(current_zeta_cfg);
```

Comment:
do we really need to lock the create and update operation using _group_operation_mutex? keep in mind ovs db already handles concurrency with its own table locks.